crates/sui-rpc-api/proto/sui.node.v2.proto

```diff
@@ -0,0 +1,352 @@
+  optional google.protobuf.Timestamp timestamp = 5;
+
+  // The lowest checkpoint for which checkpoints and transaction data is available
+  optional uint64 lowest_available_checkpoint = 6;
+
+  // The lowest checkpoint for which object data is available
+  optional uint64 lowest_available_checkpoint_objects = 7;
+
+  optional string software_version = 8;
+}
+
+message GetCommitteeRequest {
+  optional uint64 epoch = 1;
+}
+
+message GetCommitteeResponse {
+  optional sui.types.ValidatorCommittee committee = 1;
+}
+
+message GetTransactionRequest {
+  optional sui.types.Digest digest = 1;
+
+  optional GetTransactionOptions options = 3;
+}
+
+message GetTransactionOptions {
+  // Include the sui.types.Transaction message in the response.
+  //
+  // Defaults to true if not included
+  optional bool transaction = 1;
+  // Include the Transaction formatted as BCS in the response.
+  //
+  // Defaults to false if not included
+  optional bool transaction_bcs = 2;
+  // Include the set of sui.types.UserSignature's in the response.
+  //
+  // Defaults to true if not included
+  optional bool signatures = 3;
+  // Include the sui.types.TransactionEffects message in the response.
+  //
+  // Defaults to true if not included
+  optional bool effects = 4;
+  // Include the TransactionEffects formatted as BCS in the response.
+  //
+  // Defaults to false if not included
+  optional bool effects_bcs = 5;
+  // Include the sui.types.TransactionEvents message in the response.
+  //
+  // Defaults to true if not included
+  optional bool events = 6;
+  // Include the TransactionEvents formatted as BCS in the response.
+  //
+  // Defaults to false if not included
+  optional bool events_bcs = 7;
+}
+
+message GetTransactionResponse {
+  // The digest of this transaction
+  optional sui.types.Digest digest = 1;
+  optional sui.types.Transaction transaction = 2;
+  optional sui.types.Bcs transaction_bcs = 3;
+  repeated sui.types.UserSignature signatures = 4;
+  optional sui.types.TransactionEffects effects = 5;
+  optional sui.types.Bcs effects_bcs = 6;
+  optional sui.types.TransactionEvents events = 7;
+  optional sui.types.Bcs events_bcs = 8;
+  optional uint64 checkpoint = 9;
+  optional google.protobuf.Timestamp timestamp = 10;
+}
+
+message GetObjectRequest {
+  optional sui.types.ObjectId object_id = 1;
+  optional uint64 version = 2;
+
+  optional GetObjectOptions options = 3;
+}
+
+message GetObjectOptions {
+  // Include the sui.types.Object message in the response.
+  //
+  // Defaults to true if not included
+  optional bool object = 1;
+  // Include the Object formatted as BCS in the response.
+  //
+  // Defaults to false if not included
+  optional bool object_bcs = 2;
+}
+
+message GetObjectResponse {
+  optional sui.types.ObjectId object_id = 1;
+  optional uint64 version = 2;
+  // The digest of this object
+  optional sui.types.Digest digest = 3;
+
+  optional sui.types.Object object = 4;
+  optional sui.types.Bcs object_bcs = 5;
```

Comment:
Given we are trying to return both bcs and proto in a lot of places, would it make sense to have a common struct to represent this cleanly. So something like: 
```// Generic wrapper for messages with BCS option
message BcsWrapper<T> {
  optional T message = 1;
  optional sui.types.Bcs bcs = 2;
}
```
And, then usage might look like:
```
message GetObjectResponse {
  optional sui.types.ObjectId object_id = 1;
  optional uint64 version = 2;
  // The digest of this object
  optional sui.types.Digest digest = 3;
  optional BcsWrapper<sui.types.Object> object = 4;
}
```

Reply:
That would be awesome but unfortunately proto doesn't support generics like that :(